app/ts/components/Transactions.tsx

```diff
@@ -48,18 +80,51 @@ export const Transactions = ({
 	appSettings: Signal<AppSettings>
 	fundingAmountMin: ReadonlySignal<bigint>
 }) => {
-	const transactions = createBundleTransactions(
-		interceptorPayload.peek(),
-		signers.peek(),
-		blockInfo.peek(),
-		appSettings.peek().blocksInFuture,
-		fundingAmountMin.peek(),
+	const transactions = computed(() =>
+		createBundleTransactions(interceptorPayload.peek(), signers.peek(), blockInfo.peek(), appSettings.peek().blocksInFuture, fundingAmountMin.peek()),
 	)
 
+	const parsedTransactions = useSignal<(FlashbotsBundleTransaction & { decoded?: JSXInternal.Element })[]>(transactions.peek())
+	const parseTransactionsCb = async () => {
+		try {
+			const uniqueAddresses = [...new Set(transactions.value.map((x) => x.transaction.to))]
+			// @TODO: Map correctly to APIs when adding custom rpc support
+			const requests = await Promise.all(
+				uniqueAddresses.map((address) =>
+					fetch(
+						`https://api${
+							appSettings.peek().relayEndpoint === MEV_RELAY_GOERLI ? '-goerli' : ''
+						}.etherscan.io/api?module=contract&action=getabi&address=${address}&apiKey=PSW8C433Q667DVEX5BCRMGNAH9FSGFZ7Q8`,
```

Comment:
maybe store the api key later somewhere?

Reply:
API key is ripped from ABINinja.